powersimdata/data_access/profile_helper.py

```diff
@@ -7,6 +7,39 @@
 from powersimdata.utility import server_setup
 
 
+def _get_profile_version(_fs, kind):
+    """Returns available raw profiles from the give filesystem
+    :param fs.base.FS _fs: filesystem instance
+    :param str kind: *'demand'*, *'hydro'*, *'solar'* or *'wind'*.
+    :return: (*list*) -- available profile version.
+    """
+    matching = [f for f in _fs.listdir(".") if kind in f]
+    return [f.lstrip(f"{kind}_").rstrip(".csv") for f in matching]
+
+
+def get_profile_version_cloud(grid_model, kind):
+    """Returns available raw profile from blob storage.
+
+    :param str grid_model: grid model.
+    :param str kind: *'demand'*, *'hydro'*, *'solar'* or *'wind'*.
+    :return: (*list*) -- available profile version.
+    """
+    bfs = fs.open_fs("azblob://besciences@profiles").opendir(f"raw/{grid_model}")
+    return _get_profile_version(bfs, kind)
+
+
+def get_profile_version_local(grid_model, kind):
+    """Returns available raw profile from local file.
+
+    :param str grid_model: grid model.
+    :param str kind: *'demand'*, *'hydro'*, *'solar'* or *'wind'*.
+    :return: (*list*) -- available profile version.
+    """
+    profile_dir = fs.path.join(server_setup.LOCAL_DIR, "raw", grid_model)
+    lfs = fs.open_fs(profile_dir)
+    return _get_profile_version(lfs, kind)
```

Comment:
Not sure whether there is an easy way to combine these two functions together and make location of the profiles a parameter.

Reply:
There might be a simplification we could do, say if we had the local and blob filesystems already created as part of some `DataAccess` object. This seemed like the simplest approach for now though.